src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
Oh, I didn't see your comment when I was typing my last comment! Yes, you're right about the static-ness; that was my mistake.

Today, I got familiar with your code . Tomorrow, I'll think about a solution and will share my findings. Get ready for a long discussion ;)